frame/support/src/migrations.rs

```diff
@@ -15,19 +15,128 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[cfg(feature = "try-runtime")]
-use crate::storage::unhashed::contains_prefixed_key;
 use crate::{
-	traits::{GetStorageVersion, NoStorageVersionSet, PalletInfoAccess, StorageVersion},
+	traits::{
+		GetStorageVersion, NoStorageVersionSet, OnRuntimeUpgrade, PalletInfoAccess, StorageVersion,
+	},
 	weights::{RuntimeDbWeight, Weight},
 };
 use impl_trait_for_tuples::impl_for_tuples;
 use sp_core::Get;
 use sp_io::{hashing::twox_128, storage::clear_prefix, KillStorageResult};
 use sp_std::marker::PhantomData;
+
 #[cfg(feature = "try-runtime")]
 use sp_std::vec::Vec;
 
+/// Make it easier to write versioned runtime upgrades.
+///
+/// VersionedRuntimeUpgrade allows developers to write migrations without worrying about checking
+/// and setting storage versions. Instead, the developer wraps their migration in this struct which
+/// takes care of version handling using best practices.
+///
+/// It takes 5 type parameters:
+/// - `From`: The version being upgraded from.
+/// - `To`: The version being upgraded to.
+/// - `Inner`: An implementation of `OnRuntimeUpgrade`.
+/// - `Pallet`: The Pallet being upgraded.
+/// - `Weight`: The runtime's RuntimeDbWeight implementation.
+///
+/// When a VersionedRuntimeUpgrades `on_runtime_upgrade` method is called, the on-chain version of
+/// the pallet is compared to `From`. If they match, `Inner::on_runtime_upgrade` is called and the
+/// pallets on-chain version is set to `To`. Otherwise, a warning is logged notifying the developer
+/// that the upgrade was a noop and should probably be removed.
+///
+/// Example:
+/// ```ignore
```

Comment:
possibly not to difficult to make this not be `ignore` and actually compile.

Reply:
are you thinking I'd use the dummy pallet and upgrades from my test file? I'd rather not use a migration from a real pallet since it creates a dependency that could break. I'm also curious what the benefit/s are of making the comment compile